app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -0,0 +1,136 @@
+;(function (Modules, root) {
+  'use strict'
+
+  var $ = root.$
+  var $window = $(root)
+
+  Modules.HighlightActiveSectionHeading = function () {
+    var self = this
+    var _hasResized = true
+    var _hasScrolled = true
+    var _interval = 50
+    var anchorIDs = []
+
+    self.getWindowDimensions = function () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      }
+    }
+
+    self.getWindowPositions = function () {
+      return {
+        scrollTop: $window.scrollTop()
+      }
+    }
+
+    self.getElementOffset = function ($el) {
+      return $el.offset()
+    }
+
+    self.start = function ($el) {
+      $window.resize(self.hasResized)
+      $window.scroll(self.hasScrolled)
+
+      setInterval(self.checkResize, _interval)
+      setInterval(self.checkScroll, _interval)
```

Comment:
Is there a reason we're polling at intervals rather than binding directly to the scroll events? If it's to throttle/debounce the scroll event, have we ruled out using e.g. https://github.com/cowboy/jquery-throttle-debounce?

Either way, it'd be good to document why we're doing this as it seems a bit odd to me and difficult to follow…